apps/src/calc/calc.js

```diff
@@ -199,26 +201,31 @@ Calc.init = function (config) {
     }
   };
 
-  var codeWorkspace = (
-    <CodeWorkspace
-      localeDirection={studioApp.localeDirection()}
-      editCode={!!level.editCode}
-      readonlyWorkspace={!!config.readonlyWorkspace}
-    />
-  );
+  // Push initial level properties into the Redux store
+  studioApp.reduxStore.dispatch(setInitialLevelProps({
+    localeDirection: studioApp.localeDirection(),
```

Comment:
I was going to suggest two modules, or at least renaming the module so it's no longer `getState().level.localeDirection` which is a little misleading.  It might not hurt to make localeDirection its own reducer too (`getState().localeDirection`) - I think it's fine for us to dispatch to the store a bunch of times before we subscribe things to it?